File: src/Conduit/Models/Comment.php
<?php

namespace Conduit\Models;

use Illuminate\Database\Eloquent\Model;


/**
 * @property integer                 id
 * @property string                  body
 * @property integer                 article_id
 * @property integer                 user_id
 * @property \Conduit\Models\User    user
 * @property \Conduit\Models\Article article
 * @property \Carbon\Carbon          created_at
 * @property \Carbon\Carbon          update_at
 */
class Comment extends Model
{

    /**
     * The attributes that are mass assignable.
     *
     * @var array
     */
    protected $fillable = [
        'body',
        'user_id',
        'article_id',
    ];

    /********************
     *  Relationships
     ********************/

    public function article()
    {
        return $this->belongsTo(Article::class);
    }

    public function user()
    {
        return $this->belongsTo(User::class);
    }
}

File: app/models/comment.go
package models

import (
	"gopkg.in/gorp.v2"
	"time"
)

type Comment struct {
	ID        int
	Body      string
	ArticleID int
	UserID    int
	CreatedAt time.Time
	UpdatedAt time.Time

	// Transient
	User    User
	Article Article
}

func (i *Comment) PreInsert(s gorp.SqlExecutor) error {
	i.CreatedAt = time.Now()
	i.UpdatedAt = i.CreatedAt
	return nil
}

func (i *Comment) PreUpdate(s gorp.SqlExecutor) error {
	i.UpdatedAt = time.Now()
	return nil
}
